### PR TITLE
[PSyIR] Creating a new variable (#625)

### PR DESCRIPTION
**Pull request by rupertford**

### CODE REVIEW

**Comment by rupertford** [comment]
Ready for first review from @sergisiso or @arporter subject to travis being happy.

File: src/psyclone/psyir/symbols/symboltable.py [hunk 1]
```diff
@@ -65,6 +66,26 @@ def __init__(self, schedule=None):
...
+        `root_name` argument is not supplied then the name is
+        generated internally, otherwise the `root_name` is used. If
+        required, an additional integer is appended to avoid clashes.
+
+        :param str root_name: the name to use when creating a new \
+        symbol name. This will be appended with an integer if the name \
+        clashes with an existing symbol name.
+
+        '''
+        if not root_name:
```
Comment (sergisiso) [inline src/psyclone/psyir/symbols/symboltable.py]: Are you aware that `st.new_symbol_name("")` will make this condition True? This is probably a good think as languages do not allow variables that start with a numerical digit , but I think it should be documented in the method docstring. And tested.

Comment (sergisiso) [reply src/psyclone/psyir/symbols/symboltable.py]: Also add a type check to the root_name parameter to make sure that a string is passed.

Comment (rupertford) [reply src/psyclone/psyir/symbols/symboltable.py]: Good points. Done.

**Comment by rupertford** [comment]
Ready for next review from @sergisiso.

**Comment by sergisiso** [comment]
@rupertford  I approved the PR but also made some minor changes (see https://github.com/stfc/PSyclone/pull/627/commits/b09e2b577d1521e020db4c19913e7712e1a05a4a), I will merge it when Travis is done if you are happy with the changes.

**Pull request merged**